includes/in-person-payments/class-wc-payments-in-person-payments-receipts-service.php

```diff
@@ -84,63 +84,52 @@ private function format_line_items( array $order ) :array {
 	 * @return void
 	 * @throws \Exception Error validating settings.
 	 */
-	private function validate_settings( $settings ) {
+	private function validate_settings( array $settings ) {
 		if ( ! $settings ) {
-			throw new \Exception( 'You must provide settings information' );
+			throw new \Exception( 'You must provide settings information.' );
 		}
 
 		if ( ! array_key_exists( 'business_name', $settings ) ) {
-			throw new \Exception( 'You must provide a business name' );
+			throw new \Exception( 'You must provide a business name.' );
 		}
 
-		$this->validate_support_info( $settings['support_info'] );
-	}
-
-	/**
-	 * Validate support info
-	 *
-	 * @param  array $support_info Support info.
-	 * @return void
-	 * @throws \Exception Error validating support info.
-	 */
-	private function validate_support_info( $support_info ) {
-		if ( ! $support_info ) {
-			throw new Exception( 'You must provide support information.' );
+		if ( ! array_key_exists( 'support_info', $settings ) ) {
+			throw new \Exception( 'You must provide support information.' );
 		}
 
-		foreach ( $support_info as $key => $value ) {
+		foreach ( $settings['support_info'] as $key => $value ) {
 			if ( ! in_array( $key, [ 'address', 'email', 'phone' ], true ) ) {
-				throw new Exception( 'Invalid support information.' );
+				throw new Exception( sprintf( 'Error validating support information. Invalid key: %s', $key ) );
 			}
 		}
```

Comment:
Consider these 2 examples:
- we pass in additional key to `$settings['support_info']`. We'll get an exception thrown even though the additional key will not brake template generation.
- we pass in empty array for `$settings['support_info']`. We won't get an exception but the template generation will fail. 

We need to reverse the order of loops
```
foreach ( [ 'address', 'email', 'phone' ] as $required_key ) {
	if ( ! in_array( $required_key, $settings['support_info'], true ) ) {
		throw new Exception( sprintf( 'Error validating support information. Missing key: %s', $required_key ) );
	}
}
```

Reply:
Thanks @KarlisJ nicer this way